From e58d67d976dc357115dd822aa8526ff4bf1e2923 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Fri, 12 Feb 2021 00:04:27 -0800
Subject: [PATCH] makefile: use sh for test output filtering

There's nothing specific to bash here, and using sh will make this
compatible with systems that don't ship with bash (e.g. Alpine Linux)
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index b9479e2..9fd6542 100644
--- a/Makefile
+++ b/Makefile
@@ -192,7 +192,7 @@ $(BDIR)/test_lurch_util: $(OBJECTS_W_COVERAGE) $(VENDOR_LIBS) $(BDIR)/test_lurch
 	-Wl,--wrap=purple_debug_info \
 	-Wl,--wrap=purple_debug_misc \
 	-Wl,--wrap=purple_base16_encode_chunked
-	bash -c "set -o pipefail; $@ 2>&1 | grep -Ev ".*CRITICAL.*" | tr -s '\n'" # filter annoying and irrelevant glib output
+	sh -c "set -o pipefail; $@ 2>&1 | grep -Ev ".*CRITICAL.*" | tr -s '\n'" # filter annoying and irrelevant glib output
 
 $(BDIR)/test_lurch_api: $(OBJECTS_W_COVERAGE) $(VENDOR_LIBS) $(BDIR)/test_lurch_api.o
 	$(CC) $(CFLAGS) $(CPPFLAGS) -O0 --coverage $^ $(PURPLE_DIR)/libjabber.so.0 -o $@ $(LDFLAGS_T) \
@@ -217,7 +217,7 @@ $(BDIR)/test_lurch_api: $(OBJECTS_W_COVERAGE) $(VENDOR_LIBS) $(BDIR)/test_lurch_
 	-Wl,--wrap=omemo_storage_chatlist_exists \
 	-Wl,--wrap=omemo_storage_user_devicelist_retrieve \
 	-Wl,--wrap=lurch_util_fp_get_printable
-	bash -c "set -o pipefail; $@ 2>&1 | grep -Ev ".*CRITICAL.*" | tr -s '\n'" # filter annoying and irrelevant glib output
+	sh -c "set -o pipefail; $@ 2>&1 | grep -Ev ".*CRITICAL.*" | tr -s '\n'" # filter annoying and irrelevant glib output
 
 test: $(OBJECTS_W_COVERAGE) $(VENDOR_LIBS) $(TEST_TARGETS)
 
-- 
2.30.1

